plutus-pab/test/full/Plutus/PAB/CliSpec.hs

```diff
@@ -0,0 +1,249 @@
+
+  -- Ensure the db is set up
+  App.migrate (convertLog (PrettyObject . PABMsg) trace) (dbConfig pabConfig)
+
+  -- Spin up the servers
+  let cmd = ForkCommands
+              [ MockNode WithMockServer
+              , ChainIndex
+              , Metadata
+              , MockWallet
+              , PABWebserver
+              ]
+
+  let mkArgs availability = ConfigCommandArgs
+                { ccaTrace = convertLog PrettyObject trace
+                , ccaLoggingConfig = logConfig
+                , ccaPABConfig = pabConfig
+                , ccaAvailability = availability
+                , ccaStorageBackend = BeamSqliteBackend
+                }
+
+  args <- mkArgs <$> newToken
+
+  let handler = Builtin.handleBuiltin @TestingContracts
+
+  async . void $ runConfigCommand handler args cmd
+
+  -- Wait for them to be started
+  threadDelay $ 10 * 1000000
+
+getClientEnv :: Config -> IO ClientEnv
+getClientEnv pabConfig = do
+  manager <- newManager defaultManagerSettings
+
+  let newApiUrl = PAB.Types.baseUrl (pabWebserverConfig pabConfig)
+
+  pure $ mkClientEnv manager newApiUrl
+
+startPingPongContract :: Config -> IO ContractInstanceId
+startPingPongContract pabConfig = do
+  apiClientEnv <- getClientEnv pabConfig
+
+  let ca = ContractActivationArgs
+                { caID     = PingPong
+                , caWallet = Wallet 1
+                }
+
+  let (activateContract :<|> instance' :<|> _) = client (Proxy @(NewAPI TestingContracts Integer))
+
+  eci <- runClientM (activateContract ca) apiClientEnv
+
+  case eci of
+    Left e   -> error $ "Error starting contract: " <> show e
+    Right ci -> pure ci
+
+
+-- | Tag whether or not we expect the calls to succeed.
+data EndpointCall = Succeed String
+                  | Fail String
+ep :: EndpointCall -> String
+ep (Succeed s) = s
+ep (Fail s)    = s
+
+runPabInstanceEndpoints :: Config -> ContractInstanceId -> [EndpointCall] -> IO ()
+runPabInstanceEndpoints pabConfig instanceId endpoints = do
+  apiClientEnv <- getClientEnv pabConfig
+
+  let activateContract :<|> instance' :<|> _ = client (Proxy @(NewAPI TestingContracts Integer))
+  let status' :<|> endpoint :<|> stop' = instance' . Text.pack . show . unContractInstanceId $ instanceId
+
+  forM_ endpoints $ \e -> do
+    x <- runClientM (endpoint (ep e) (toJSON ())) apiClientEnv
+    case e of
+      Succeed _ -> do
+          assertEqual "Got the right thing back from the API" (Right ()) x
+      Fail _ -> do
+          assertBool "Endpoint call succeeded (it should've failed.)" (isLeft x)
+
+
+restoreContractStateTests :: TestTree
+restoreContractStateTests =
+  testGroup "restoreContractState scenarios"
+    [ testCase "Can init,pong,ping in one PAB instance" $ do
+        -- This isn't testing anything related to restoring state; but simply
+        -- provides evidence that if the subsequent tests _fail_, then that is
+        -- an genuine error.
+        pabConfig <- decodeFileThrow @IO "./test/full/testing-plutus-pab.yaml"
```

Comment:
Instead of reading the config from a file, we could just define a value of `Config`. It would mean one less .yaml file to update.

Reply:
@j-mueller That sounds reasonable; will do.